### PR TITLE
Bump Go dependencies

### PR DESCRIPTION
**Pull request by codingllama**
Update minor/patch versions of direct dependencies.

Changelogs (minor bumps):

* https://github.com/googleapis/google-cloud-go/blob/main/CHANGES.md
* https://github.com/aws/aws-sdk-go-v2/blob/main/config/CHANGELOG.md
* https://github.com/aws/aws-sdk-go-v2/blob/main/credentials/CHANGELOG.md
* https://github.com/googleapis/gax-go/blob/main/v2/CHANGES.md
* https://github.com/mdlayher/netlink/blob/main/CHANGELOG.md

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ updates:
       # Deprecated APIs, requires manual changes.
       # TODO(xacrimon): Update Firestore and solve deprecations.
       - dependency-name: cloud.google.com/go/firestore
+      # TODO(@espadolini): Update mongo-driver.
+      - dependency-name: go.mongodb.org/mongo-driver
```
Comment (codingllama) [inline .github/dependabot.yml:12]: FYI @espadolini.

**Comment by codingllama** [comment]
FYI @nklaassen, manual deps update.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@ updates:
       # TODO(xacrimon): Update Firestore and solve deprecations.
       - dependency-name: cloud.google.com/go/firestore
       # Breaks backwards compatibility
+      - dependency-name: go.mongodb.org/mongo-driver
```
Comment (codingllama) [inline .github/dependabot.yml:12]: FYI @jakule @greedy52, could one of you take a look at updating mongo-driver when you have the time?  (Assuming I got the right people this time.)

Comment (greedy52) [reply .github/dependabot.yml:12]: there is some refactoring in the upstream driver. will do a PR to update it but likely it will take some time.

Comment (codingllama) [reply .github/dependabot.yml:12]: Thanks. Please remove it from the ignore list when you do.

**Review by nklaassen — APPROVED** [review]

**Review by jakule — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (greedy52) [reply .github/dependabot.yml]: I've made a [change](https://github.com/gravitational/teleport/pull/18646/files) to update to the new driver. Unfortunately, [this release v1.11.0](https://github.com/mongodb/mongo-go-driver/releases/tag/v1.11.0) deprecates Mongo versions older than v3.6, making all these tests fail:
https://github.com/gravitational/teleport/blob/7b11fbf3b8213a9fb4a963b58ce95e7632499832/lib/srv/db/access_test.go#L776-L779
with error
```
connection() error occurred during connection handshake: command failure: {"$err": "error connecting to the database\n\tserver at localhost:54395 reports wire version 5, but this version of the Go driver requires at least 6 (MongoDB 3.6)"}
```
I am leaning towards keeping the old driver. Not sure how many people are still on old Mongo though. @smallinsky what do you think?

Comment (smallinsky) [reply .github/dependabot.yml]: Definitely we should not drop the support for old versions -  probably some of our customer is using it. Recently we got issue related to mongoDB 3.6 https://github.com/gravitational/teleport/issues/18549 that confirm that old mongoDB versions are still in use.

Comment (codingllama) [reply .github/dependabot.yml]: Raised https://github.com/gravitational/teleport/issues/18777 to track this. I think it's wise to plan a way forward, so we are not caught by surprise by this when a driver update becomes necessary.

@greedy52, once we know what the plan is, please update the TODO in dependabot.yml accordingly.

**Pull request merged**